algo/src/test/java/com/hartwig/serve/sources/ckb/CkbVariantAnnotatorTest.java

```diff
@@ -27,40 +34,78 @@ public class CkbVariantAnnotatorTest {
+        KnownCopyNumber annotated = CkbVariantAnnotator.annotateCopyNumber(copyNumber, variantWith(UNKNOWN_GENE, "no effect"));
+        assertEquals(GeneRole.UNKNOWN, annotated.geneRole());
+        assertEquals(ProteinEffect.NO_EFFECT, annotated.proteinEffect());
+        assertEquals(true, annotated.associatedWithDrugResistance());
+    }
+
+    @Test
+    public void shouldAnnotateFusionUsingVariant() {
+        KnownFusion exon = FusionTestFactory.knownFusionBuilder().build();
+        KnownFusion annotated = CkbVariantAnnotator.annotateFusion(exon, variantWith(BOTH_GENE, "unknown"));
+        assertEquals(ProteinEffect.UNKNOWN, annotated.proteinEffect());
+        assertEquals(true, annotated.associatedWithDrugResistance());
+    }
+
     @Test
     public void shouldAnnotateKnownGenesWithRoleFromVariant() {
-        verifyKnownGeneAnnotatedWith(ONCO_GENE, GeneRole.ONCO);
-        verifyKnownGeneAnnotatedWith(TSG_GENE, GeneRole.TSG);
-        verifyKnownGeneAnnotatedWith(BOTH_GENE, GeneRole.BOTH);
-        verifyKnownGeneAnnotatedWith(UNKNOWN_GENE, GeneRole.UNKNOWN);
+        assertGeneRoleResolved(ONCO_GENE, GeneRole.ONCO);
+        assertGeneRoleResolved(TSG_GENE, GeneRole.TSG);
+        assertGeneRoleResolved(BOTH_GENE, GeneRole.BOTH);
+        assertGeneRoleResolved(UNKNOWN_GENE, GeneRole.UNKNOWN);
     }
 
     @NotNull
     private static ImmutableGene geneWith(String role) {
         return ImmutableGene.builder().id(0).createDate(TEST_DATE).updateDate(TEST_DATE).geneSymbol(GENE_NAME).geneRole(role).build();
     }
 
-    private static void verifyKnownGeneAnnotatedWith(Gene gene, GeneRole expectedRole) {
-        ExtractionResult result = CkbVariantAnnotator.annotate(ImmutableExtractionResult.builder()
-                .refGenomeVersion(RefGenome.V37)
-                .addKnownGenes(ImmutableKnownGene.builder().gene(GENE_NAME).geneRole(GeneRole.UNKNOWN).build())
-                .build(), variantWith(gene));
-        assertEquals(result.knownGenes().size(), 1);
-        KnownGene annotated = result.knownGenes().iterator().next();
-        assertEquals(annotated.gene(), GENE_NAME);
-        assertEquals(annotated.geneRole(), expectedRole);
+    private static void assertGeneRoleResolved(Gene gene, GeneRole expectedRole) {
+        Variant variant = variantWith(gene, null);
+        assertEquals(expectedRole, CkbVariantAnnotator.resolveGeneRole(variant));
     }
 
     @NotNull
-    private static ImmutableVariant variantWith(Gene gene) {
+    private static ImmutableVariant variantWith(@NotNull Gene gene, @Nullable String proteinEffect) {
         return ImmutableVariant.builder()
                 .id(0)
                 .createDate(TEST_DATE)
                 .updateDate(TEST_DATE)
                 .fullName("variant")
                 .variant("variant")
                 .isHotspot(false)
-                .gene(gene)
+                .gene(gene).proteinEffect(proteinEffect).associatedWithDrugResistance("Y")
```

Comment:
Not a comment but just a remark: Nearly all actual variants have `associatedWithDrugResistance` null, seeing `Y` as default looks weird to me but I guess a test doesn't have to reflect reality :)

Also - FYI: There is no `N` value here, since CKB can never definitively know whether a variant implies drug resistance for any of the existing or future drugs!